sdk/ml/azure-ai-ml/tests/pipeline_job/e2etests/test_pipeline_job.py

```diff
@@ -1344,6 +1344,133 @@ def test_pipeline_node_with_default_component(self, client: MLClient, randstr: C
             == "microsoftsamples_command_component_basic@default"
         )
 
+    def test_register_output_without_name_cli(self, client: MLClient):
+        register_pipeline_output_path = "tests/test_configs/pipeline_jobs/helloworld_pipeline_job_register_pipeline_output_without_name.yaml"
+        register_node_output_path = "tests/test_configs/pipeline_jobs/helloworld_pipeline_job_register_node_output_without_name.yaml"
+        error_message_pipeline = "PipelineOutput.name is missing."
+        error_message_node = "NodeOutput.name is missing."
+
+        with pytest.raises(UserErrorException) as e:
+            pipeline = load_job(source=register_pipeline_output_path)
```

Comment:
put the load_job outside of the with.
The error is raised during create_or_update?